randomActivityGen.py

```diff
@@ -28,15 +33,74 @@
 
 import sumolib
 
+
+def find_city_centre(net: sumolib.net.Net) -> Tuple[float, float]:
+    """
+    Finds the city centre; average node coord of all nodes in the net
+    """
+    node_coords = [node.getCoord() for node in net.getNodes()]
+    return float(np.mean([c[0] for c in node_coords])), float(np.mean([c[1] for c in node_coords]))
+
+
+def setup_city_gates(net: sumolib.net.Net, stats: ET.ElementTree, gate_count: int):
+    assert gate_count >= 0, "Number of city gates cannot be negative"
+    # Find existing gates to determine how many we need to insert
+    xml_gates = stats.find("cityGates")
+    xml_entrances = xml_gates.findall("entrance")
+    n = gate_count - len(xml_entrances)
+    if n < 0:
+        print(f"Warning: {gate_count} city gate was requested, but there are already {len(xml_entrances)} defined")
+    if n <= 0:
```

Comment:
Would it not make more sense that it is the if statement on line 51 that is n <= 0? 
Does not make sense to insert 0 new city gates

Reply:
If 0 gates needs to be inserted the function returns on line 54. It is not unreasonable to want 0 gates in the city. But if the user wants less gates than already defined in the stats file then we will warn the user (and also return on line 54).

Reply:
I see, I guess it confused me that line 53 says n <= 0, when its simply a check if n == 0?

Reply:
But it's not. We warn on negative and abort on negative and zero.
Anyway, it's merged.